process/block/baseProcess.go

```diff
@@ -1693,6 +1692,30 @@ func gasAndFeesDelta(initialGasAndFees, finalGasAndFees scheduled.GasAndFees) sc
 	}
 }
 
+// For the expected results, this method should not be called anymore after the call of method ProcessScheduledBlock,
+// as the executed scheduled mini blocks will be overwritten with those for the new block
+func (bp *baseProcessor) getIndexOfFirstMiniBlockToBeExecuted(header data.HeaderHandler, body *block.Body) int {
+	if !bp.flagScheduledMiniBlocks.IsSet() {
+		return 0
+	}
+
+	var mbIndex int
+	for index := range body.MiniBlocks {
+		mbHash := header.GetMiniBlockHeadersHashes()[index]
+		if bp.scheduledTxsExecutionHandler.IsMiniBlockExecuted(mbHash) {
```

Comment:
Yep, this will be done by the proposer by setting in the mini block header IsFinal property. I will add also here a TODO to be refactored after we have the last version in the elrond-go-core repository

Reply:
Actually here is a little complicated to replace this approach with IsFinal, as also all the mbs from the current body, last scheduled mbs (which are now added here as final) + normal mbs from the current block (which are not yet executed) will have isFinal call return as true. So this method will not work correctly any more with this change.